app/controllers/api/v2/courses_controller.rb

```diff
@@ -22,33 +22,37 @@ def show
       end
 
       def sync_with_search_and_compare
-        response = sync_courses
-
-        if response
-          head :ok
-        else
-          raise RuntimeError.new(
-            'error received when syncing with search and compare'
+        if @course.syncable?
+          response = SearchAndCompareAPIService::Request.sync(
+            [@course]
           )
+          if response
+          else
+            raise RuntimeError.new(
+              'error received when syncing with search and compare'
+            )
+          end
         end
       end
 
       def publish
         if @course.publishable?
           @course.publish_sites
           @course.publish_enrichment(@current_user)
-
-          response = ManageCoursesAPIService::Request.sync_course_with_search_and_compare(
-            @current_user.email,
-            @provider.provider_code,
-            @course.course_code
-          )
-
-          if response
-            head :ok
+          if @course.syncable?
```

Comment:
maybe call `sync_with_search_and_compare`

Reply:
Is this what you were after?

Reply:
yes, and you can drop `if @course.syncable?`

Reply:
Cool. All done 👍